### PR TITLE
overlord/devicestate: try to fetch/refresh the signing key of serial (also in case is not there yet)

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

**Comment by pedronis** [comment]
based on #1806

**Comment by matiasb** [comment]
Makes sense to me.

File: overlord/devicestate/devicemgr.go [hunk 1]
```diff
@@ -139,13 +141,24 @@ func (m *DeviceManager) Stop() {
...
 
+func useStaging() bool {
+	return os.Getenv("SNAPPY_USE_STAGING_STORE") == "1"
+}
+
+func deviceAPIBaseURL() string {
+	if useStaging() {
+		return "https://myapps.developer.staging.ubuntu.com/identity/api/v1/"
+	}
+	return "https://myapps.developer.ubuntu.com/identity/api/v1/"
```
Comment (zyga) [inline overlord/devicestate/devicemgr.go]: It would be good to be able to override this with a totally custom URL.

**Comment by zyga** [comment]
Read half of it (didn't touch tests yet) and while I don't feel comfortable with assertions fully the changes made sense to me.

**Pull request merged**